pkg/skaffold/build/docker/docker.go

```diff
@@ -54,7 +54,7 @@ func (b *Builder) Build(ctx context.Context, out io.Writer, a *latestV1.Artifact
 
 	var imageID string
 
-	if b.useCLI || b.useBuildKit {
+	if b.useCLI || (b.useBuildKit != nil && *b.useBuildKit) {
```

Comment:
Worth a comment here to say that we ignore the useCLI setting if buildkit is requested as we currently use the Docker CLI to use buildkit. We could consider a different approach in the future.